pom.xml

```diff
@@ -1383,6 +1383,40 @@
         </dependency>
       </dependencies>
     </profile>
+    <profile>
+      <id>java17</id>
+      <activation>
+        <jdk>[17,)</jdk>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-surefire-plugin</artifactId>
+            <configuration>
+              <argLine>--add-opens java.base/java.lang=ALL-UNNAMED</argLine>
```

Comment:
argLine should also contain the -**_Xmx1024m -XX:+HeapDumpOnOutOfMemoryError_**, otherwise it will get overwritten. This can be confirmend by running test in debug mode. Rest of the configuration, evn and sytem properties are carried forward to java17 profile from the initial maven surefire plugin 
```
mvn test -Dtest="org.apache.tez.dag.history.recovery.TestRecoveryService#testRecoveryFlushOnSummaryEvent" -X | grep "argLine"
```